README.md

```diff
@@ -6,46 +6,71 @@ Perform work across multiple GitHub repos. Useful for simple batch updates & bat
 yarn add polymer-workspaces
 ```
 
-## Usage Example: Initialization
+## Example: Creating a New Workspace
 
 ```js
 const {Workspace} = require('polymer-workspace');
 const path = require('path');
 
-const ws = new Workspace({
-  // get a GitHub API token: https://github.com/blog/1509-personal-api-tokens
+const workspace = new Workspace({
+  // Get a GitHub API token: https://github.com/blog/1509-personal-api-tokens
   token: 'GITHUB_API_TOKEN',
-  // and choose a "workspace" directory
+  // Choose a "workspace" directory
   dir: path.resolve(process.cwd(), '.workspace');
 });
 
-const initializedRepos = await ws.init({
+// Check out & set up the given repos from GitHub.
+const workspaceRepos = await workspace.init({
```

Comment:
Feedback for possibly a future API change, but consider it for this refactor:

`init` methods are always worth looking at closely to see if they can go away in favor of the constructor. `init` implies it can only be called once, like the constructor.

Also here, I wonder if the set of repo's can change over time, ie. can `init` actually be called again? If not, it might be better to fold this into the constructor. We may like the idea of a constructor that doesn't do I/O, and so keep `init`, but with no arguments.

And since a Workspace remembers the list of repos, I think an API improvement would be to move the `run` method to `Workspace` so that the following example is more like:

```js
const {successes, failures} = await Workspace.forEach(async(repo) => {
  ...
});
```

Breaking out the other operations makes sense, but keeping one core task running method that abstracts a little over passing around the list of repos seems nice.

Reply:
You're right, agreed that init() & constructor have a lot of overlap. It was a secondary (aspirational) goal to be able to `init()` multiple times, but there's no real use case for that that I know of. Folding these options into the constructor and simplifying the number of situations `Workspace` needs to handle makes sense.

Re `Workspace.forEach`, we talked in person, but `run()` actually makes a ton of sense to be pulled out. For complex changes, you may want to run several different batches of work in a row. The consumer should have control over which repos each step gets run over. In the modulizer, for example, when an operation fails for one repo we report and then keep that repo out of future batches to prevent Very Very Bad Unexpected Things. 

```ts
const results1 = run(allRepos, /* ... */);
const results2 = run(results1.successes, /* ... */);
const results3 = run(results2.successes, /* ... */);
```

This also makes sense conceptually, because it means that `Workspace` can focus on initialization and setup exclusively.